Include/Rover/Range.hpp

```diff
@@ -80,11 +80,15 @@ namespace Rover {
     private:
       Type calculate_random(const Type& begin, const Type& end);
       Type pick_random(const Type& lhs, const Type& rhs);
+      template<typename TypeFwd>
+      Type round(Evaluator& evaluator, TypeFwd&& value);
+      using GranularityPlaceholder = std::conditional_t<std::is_same_v<
```

Comment:
Somewhat ugly, and the choice of an optimal type depends on alignment, but I haven't really found a more effective way.